aptos-move/framework/src/natives/transaction_context.rs

```diff
@@ -34,6 +40,43 @@ impl NativeTransactionContext {
     }
 }
 
+/***************************************************************************************************
+ * native fun create_uuid
+ *
+ *   gas cost: base_cost
+ *
+ **************************************************************************************************/
+#[derive(Clone, Debug)]
+pub struct CreateUUIDGasParameters {
+    pub base: InternalGas,
+}
+
+fn native_create_uuid(
+    gas_params: &CreateUUIDGasParameters,
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(gas_params.base)?;
+
+    let mut transaction_context = context
+        .extensions_mut()
+        .get_mut::<NativeTransactionContext>();
+    transaction_context.uuid_counter += 1;
+    const OBJECT_FROM_TRANSACTION_UUID_ADDRESS_SCHEME: u8 = 0xFB;
```

Comment:
This domain separator **MUST NOT** be a dangling hardcoded constant here. Otherwise, it will be very difficult to account for when adding new things later on that need to be domain separated.

What are you domain-separating from here? The other address derivation methods from [here](https://github.com/aptos-labs/aptos-core/blob/main/types/src/transaction/authenticator.rs#L241)?

Reply:
@alinush - it would be great to "domain separate it from everything that uses txn_hash", so it can be generally used (instead of having generate_uuid_for_object/generate_uuid_for_account/generate_uuid_for_x)


for this PR it needs to be separated from these : https://github.com/aptos-labs/aptos-core/blob/d65effee8c96b61e2e824a3865e091e6ac220adf/aptos-move/framework/aptos-framework/sources/object.move#L65

Reply:
That's just this [constant from here](https://github.com/aptos-labs/aptos-core/blob/main/types/src/transaction/authenticator.rs#LL250C5-L250C32) hardcoded in Move because there's no other way to export it.

This means that @vusirikala should move his `OBJECT_FROM_TRANSACTION_UUID_ADDRESS_SCHEME` constant into the [`Scheme` struct from that file](https://github.com/aptos-labs/aptos-core/blob/main/types/src/transaction/authenticator.rs#L241) and read it from there.

This will be easy to track.